Gems/RemoteTools/Code/Source/RemoteToolsSystemComponent.h

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzFramework/Network/IRemoteTools.h>
+#include <AzNetworking/ConnectionLayer/IConnectionListener.h>
+#include <AzNetworking/DataStructures/ByteBuffer.h>
+#include <AzNetworking/Utilities/TimedThread.h>
+
+#include "Utilities/RemoteToolsJoinThread.h"
+#include "Utilities/RemoteToolsOutboxThread.h"
+
+namespace RemoteToolsPackets
+{
+    class RemoteToolsConnect;
+    class RemoteToolsMessage;
+} // namespace RemoteToolsPackets
+
+namespace RemoteTools
+{
+    // Slightly below AzNetworking's TCP max packet size for maximum message space with room for packet headers
+    static constexpr uint32_t RemoteToolsBufferSize = 16000;
```

Comment:
is there a tcp max size variable you could use here? like TCP::MaxPacketSize - 1000; so you know it really is below the max tcp size?